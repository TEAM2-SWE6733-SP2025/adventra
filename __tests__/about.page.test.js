import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import AboutPage from "../src/app/about/page";

jest.mock("../src/app/components/Navbar.jsx", () => {
  const MockNavbar = () => <nav>Mock Navbar</nav>;
  MockNavbar.displayName = "MockNavbar";
  return MockNavbar;
});

describe("AboutPage", () => {
  it("renders the About Us heading", () => {
    render(<AboutPage />);

    const heading = screen.getByRole("heading", {
      name: /Welcome to Adventra!/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it("renders the welcome message", () => {
    render(<AboutPage />);

    const welcomeMessage = screen.getByText(/At Adventra/i);

    expect(welcomeMessage).toBeInTheDocument();
  });

  it("renders the mission statement", () => {
    render(<AboutPage />);

    const missionStatement = screen.getByText(
      /our platform connects like-minded adventurers/i
    );

    expect(missionStatement).toBeInTheDocument();
  });
});
