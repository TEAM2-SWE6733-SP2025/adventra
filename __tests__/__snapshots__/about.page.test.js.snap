// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="block w-full"
  >
    <nav>
      Mock Navbar
    </nav>
  </header>
  <div
    class="min-h-screen flex items-center justify-center p-8"
  >
    <div
      class="max-w-4xl w-full"
    >
      <h1
        class="text-4xl font-extrabold text-yellow-500 mb-8"
      >
        Welcome to Adventra!
      </h1>
      <p
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        At Adventra, we believe that life is best experienced through shared adventures. Whether you're into hiking, skiing, backpacking, or simply exploring new places, our platform connects like-minded adventurers. Our goal is to help you find your perfect adventure buddy — someone who matches your skills, preferences, and adventurous spirit.
      </p>
      <p
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        With a passionate team led by experienced developers, product innovators, and design experts, we are committed to making every journey memorable. From planning your next escapade to capturing unforgettable moments, Adventra is your go-to travel companion.
      </p>
      <p
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        Let’s embark on your next adventure together. Your story starts here.
      </p>
      <h2
        class="text-4xl font-extrabold text-yellow-500 mb-8"
      >
        Meet Our Team
      </h2>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-8"
      >
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <div
            class="w-24 h-24 bg-yellow-500 text-black font-bold flex items-center justify-center rounded-full text-2xl mb-4"
          >
            K
          </div>
          <h2
            class="text-xl font-semibold"
          >
            Kirsten Huttar
          </h2>
          <p
            class="text-gray-400"
          >
            Scrum Master
          </p>
        </div>
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <a
            href="/profilepic.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="John Pranoy Yalla"
              class="w-24 h-24 rounded-full mb-4 object-cover border-4 border-yellow-500"
              data-nimg="1"
              decoding="async"
              height="96"
              loading="lazy"
              src="/_next/image?url=%2Fprofilepic.png&w=256&q=75"
              srcset="/_next/image?url=%2Fprofilepic.png&w=96&q=75 1x, /_next/image?url=%2Fprofilepic.png&w=256&q=75 2x"
              style="color: transparent;"
              width="96"
            />
          </a>
          <h2
            class="text-xl font-semibold"
          >
            John Pranoy Yalla
          </h2>
          <p
            class="text-gray-400"
          >
            Product Owner
          </p>
        </div>
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <a
            href="/team-1.jpg"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Louis Muhammad"
              class="w-24 h-24 rounded-full mb-4 object-cover border-4 border-yellow-500"
              data-nimg="1"
              decoding="async"
              height="96"
              loading="lazy"
              src="/_next/image?url=%2Fteam-1.jpg&w=256&q=75"
              srcset="/_next/image?url=%2Fteam-1.jpg&w=96&q=75 1x, /_next/image?url=%2Fteam-1.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="96"
            />
          </a>
          <h2
            class="text-xl font-semibold"
          >
            Louis Muhammad
          </h2>
          <p
            class="text-gray-400"
          >
            Senior Developer
          </p>
        </div>
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <div
            class="w-24 h-24 bg-yellow-500 text-black font-bold flex items-center justify-center rounded-full text-2xl mb-4"
          >
            C
          </div>
          <h2
            class="text-xl font-semibold"
          >
            Christopher Wilder
          </h2>
          <p
            class="text-gray-400"
          >
            Developer
          </p>
        </div>
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <div
            class="w-24 h-24 bg-yellow-500 text-black font-bold flex items-center justify-center rounded-full text-2xl mb-4"
          >
            P
          </div>
          <h2
            class="text-xl font-semibold"
          >
            Prudhvi Samudrala
          </h2>
          <p
            class="text-gray-400"
          >
            Developer
          </p>
        </div>
        <div
          class="p-6 dark rounded-2xl shadow-lg flex flex-col items-center"
        >
          <div
            class="w-24 h-24 bg-yellow-500 text-black font-bold flex items-center justify-center rounded-full text-2xl mb-4"
          >
            Y
          </div>
          <h2
            class="text-xl font-semibold"
          >
            Yaswitha Swarna
          </h2>
          <p
            class="text-gray-400"
          >
            UI / Tester
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
